### ISSUE TITLE
ModelWasBanned doesn't fire - here's why.

### ISSUE THREAD

**Issue by rickmacgillis** [cybercog/laravel-ban#44]
So, today I ran into a problem that caused me to have to solve the issue with some of Laravel Ban's events not firing. I found out that only in the `created` method of `BanObserver`, if you try to save the bannable, then it stops the event from firing. Likewise, if that event fires, then it also stops the method from running to completion.

The only thing I can think of that would do something like that is `exit` or some kind of exception or error. The problem is that I never use `exit` in my own code. (I even did a file search on all of my code, and nothing uses that language construct.)

In my dev environment, I have it set to log all of the errors and exceptions to a single file. It doesn't show any. Plus, Laravel uses `filp/Whoops` (lol) to display exceptions that crash the system. None of that happened.

I've spent hours trying to debug this issue, and I'm at an impasse. Any chance you can have a go at it and make the event start firing for everyone?

**Comment by antonkomarev** [comment]
Hi, @rickmacgillis! That's seems really strange for me. I will be able too look on it a bit later.
Could it be reproduced in tests or sandbox?

**Comment by rickmacgillis** [comment]
Presumably this issue is related to my old issue #34, but now it doesn't trigger in the live environment. The tests from the old issue still don't work. (Neither the banned or unbanned events)

As for "sandbox", what are you referring to?

**Comment by antonkomarev** [comment]
Thank you for the previous issue link!

As sandbox I meant github public repo with minimum required functionality to reproduce the bug.

**Comment by rickmacgillis** [comment]
I'll give it a go tomorrow when I'm back in the office.

⁣Rick Mac Gillis
Inventor​

**Comment by rickmacgillis** [comment]
I'm working on this issue now. Once I created the sandbox it worked fine with the tests, so I'm trying to find out what's messing things up. It's not Laravel Spark.

I did find out that my test was flawed for `ModelWasUnbanned`. To make that event trigger, the model needs to be banned first. (That's a good thing that duplicate unbans don't trigger it.)

I'll keep you updated to see what comes of the issue.
